igvm/hypervisor_preferences.py

```diff
@@ -194,10 +194,11 @@ class to let them be prepared lazily.  When Python needs to compare
         for index, comparable in enumerate(comparables):
             if not comparable.executed:
                 break
+        else:
+            index += 1
```

Comment:
Just to make me remember, I see three scenarios here:
- The loop ran e.g. 3 times with indexes from 0-2 and we hit the break on index==2. We won't hit the else as we used break. index is now 2, which is correct. ✅
- The loop ran e.g. 3 times with indexes from 0-2 and we never hit the break. We will hit the else. Index is now 3, which is correct. ✅
- The loop runs 0 times if enumerate(comparables) is empty to begin with. We will hit the else. Index would be undefined here and this would throw an NameError. ❌

We would have hit the same NameError before once we hit the logger call below, so I won't make you fix this if you don't care. Can't happen for us in production currently anyway.

Reply:
Do you think it's worth changing this to?:
```python
try:
    index += 1
except NameError:
    index = 0
```

Reply:
`comparables` cannot really be empty, but I've fixed it anyway.  `len()` is maybe also slightly more self-explanatory.

Reply:
Ah, that's a great idea. +1